sql: TestLogic/local/pg_catalog fails under stress

On 4ea464d8376402e55bf85d4e69efd4abba33acdd using `./dev test pkg/sql/logictest --filter TestLogic/local/pg_catalog --stress -- --test_arg -show-diff`

```
    --- FAIL: TestLogic/local (0.00s)
        --- FAIL: TestLogic/local/pg_catalog (10.25s)
            logic.go:2493: 
                 
                /private/var/tmp/_bazel_ssd/a51ea94df2f483aee3088dde6a3dde4e/sandbox/darwin-sandbox/4806/execroot/cockroach/bazel-out/darwin-fastbuild/bin/pkg/sql/logictest/logictest_test_/logictest_test.runfiles/cockroach/pkg/sql/logictest/testdata/logic_test/pg_catalog:1407: SELECT classid, objid, objsubid, refclassid, refobjid, refobjsubid, deptype
                FROM pg_catalog.pg_depend
                ORDER BY objid
                expected:
                    classid     objid       objsubid  refclassid  refobjid    refobjsubid  deptype
                    4294967126  111         0         4294967129  110         14           a
                    4294967126  112         0         4294967129  110         15           a
                    4294967126  192087236   0         4294967129  0           0            n
                    4294967083  842401391   0         4294967129  110         1            n
                    4294967083  842401391   0         4294967129  110         2            n
                    4294967083  842401391   0         4294967129  110         4            n
                    4294967083  842401391   0         4294967129  110         3            n
                    4294967126  2061447344  0         4294967129  3687884464  0            n
                    4294967126  3764151187  0         4294967129  0           0            n
                    4294967126  3836426375  0         4294967129  3687884465  0            n
                    
                but found (query options: "colnames") :
                    classid     objid       objsubid  refclassid  refobjid    refobjsubid  deptype
                    4294967126  111         0         4294967129  110         14           a
                    4294967126  112         0         4294967129  110         15           a
                    4294967126  192087236   0         4294967129  0           0            n
                    4294967083  842401391   0         4294967129  110         1            n
                    4294967083  842401391   0         4294967129  110         2            n
                    4294967083  842401391   0         4294967129  110         3            n
                    4294967083  842401391   0         4294967129  110         4            n
                    4294967126  2061447344  0         4294967129  3687884464  0            n
                    4294967126  3764151187  0         4294967129  0           0            n
                    4294967126  3836426375  0         4294967129  3687884465  0            n
                    
                Diff:
                --- Expected
                +++ Actual
                @@ -6,4 +6,4 @@
                     4294967083  842401391   0         4294967129  110         2            n
                +    4294967083  842401391   0         4294967129  110         3            n
                     4294967083  842401391   0         4294967129  110         4            n
                -    4294967083  842401391   0         4294967129  110         3            n
                     4294967126  2061447344  0         4294967129  3687884464  0            n
            logic.go:3433: 
                /private/var/tmp/_bazel_ssd/a51ea94df2f483aee3088dde6a3dde4e/sandbox/darwin-sandbox/4806/execroot/cockroach/bazel-out/darwin-fastbuild/bin/pkg/sql/logictest/logictest_test_/logictest_test.runfiles/cockroach/pkg/sql/logictest/testdata/logic_test/pg_catalog:1424: error while processing
            logic.go:3433: /private/var/tmp/_bazel_ssd/a51ea94df2f483aee3088dde6a3dde4e/sandbox/darwin-sandbox/4806/execroot/cockroach/bazel-out/darwin-fastbuild/bin/pkg/sql/logictest/logictest_test_/logictest_test.runfiles/cockroach/pkg/sql/logictest/testdata/logic_test/pg_catalog:1424: too many errors encountered, skipping the rest of the input
    logic.go:3667: -- test log scope end --
```

--- a/pkg/sql/logictest/testdata/logic_test/pg_catalog
+++ b/pkg/sql/logictest/testdata/logic_test/pg_catalog
@@ -1407,7 +1407,7 @@ fk         {2}      NULL       NULL       NULL       NULL       NULL    NULL
 query OOIOOIT colnames
 SELECT classid, objid, objsubid, refclassid, refobjid, refobjsubid, deptype
 FROM pg_catalog.pg_depend
-ORDER BY objid
+ORDER BY objid, refobjid, refobjsubid
 ----
 classid     objid       objsubid  refclassid  refobjid    refobjsubid  deptype
 4294967126  111         0         4294967129  110         14           a
@@ -1415,8 +1415,8 @@ classid     objid       objsubid  refclassid  refobjid    refobjsubid  deptype
 4294967126  192087236   0         4294967129  0           0            n
 4294967083  842401391   0         4294967129  110         1            n
 4294967083  842401391   0         4294967129  110         2            n
-4294967083  842401391   0         4294967129  110         4            n
 4294967083  842401391   0         4294967129  110         3            n
+4294967083  842401391   0         4294967129  110         4            n
 4294967126  2061447344  0         4294967129  3687884464  0            n
 4294967126  3764151187  0         4294967129  0           0            n
 4294967126  3836426375  0         4294967129  3687884465  0            n

